servatrice/src/servatrice_database_interface.cpp

```diff
@@ -120,7 +120,27 @@ bool Servatrice_DatabaseInterface::execSqlQuery(QSqlQuery *query)
 
 bool Servatrice_DatabaseInterface::usernameIsValid(const QString &user)
 {
-    static QRegExp re = QRegExp("[a-zA-Z0-9_\\.-]+");
+    int maxNameLength = settingsCache->value("users/maxnamelength", 12).toInt();
+    int minNameLength = settingsCache->value("users/minnamelength", 6).toInt();
+    if (user.length() < minNameLength || user.length() > maxNameLength)
+        return false;
+
+    bool allowPunctuationPrefix = settingsCache->value("users/allowpunctuationprefix", false).toBool();
+    QString allowedPunctuation = settingsCache->value("users/allowedpunctuation", "_").toString();
+    if (!allowPunctuationPrefix && allowedPunctuation.contains(user.at(0)))
+        return false;
+
+    QString regEx("[");
+    if (settingsCache->value("users/allowlowercase", true).toBool())
+        regEx.append("a-z");
+    if (settingsCache->value("users/allowuppercase", true).toBool())
+        regEx.append("A-Z");
+    if(settingsCache->value("users/allownumerics", true).toBool())
+        regEx.append("0-9");
+    regEx.append(allowedPunctuation);
```

Comment:
I mean that we need to regEx.append(QRegexp::escape(allowedPunctuation)) here.
Otherway a configuration like:```allowedpunctuation=_.-]``` will lead to a bad regexp:
```
(gdb) printq5string regEx
[a-zA-Z0-9_.-]]+
```
 that will return false for any possible nickname.